mkvcore/interceptor.go

```diff
@@ -65,16 +66,70 @@ const (
 	BlockSorterWriteOutdated
 )
 
+// MultiTrackBlockSorterOption configures a MultiTrackBlockSorterOptions.
+type MultiTrackBlockSorterOption func(*MultiTrackBlockSorterOptions) error
+
+// MultiTrackBlockSorterOptions stores options for BlockWriter.
+type MultiTrackBlockSorterOptions struct {
+	maxDelayedPackets int
+	rule              BlockSorterRule
+	maxTimescaleDelay int64
+}
+
+// WithMaxDelayedPackets set the maximum number of packets that may be delayed
+// within each track.
+func WithMaxDelayedPackets(maxDelayedPackets int) MultiTrackBlockSorterOption {
+	return func(o *MultiTrackBlockSorterOptions) error {
+		o.maxDelayedPackets = maxDelayedPackets
+		return nil
+	}
+}
+
+// WithSortRule set the sort rule to apply to how packet ordering should be
+// treated within the webm container.
+func WithSortRule(rule BlockSorterRule) MultiTrackBlockSorterOption {
+	return func(o *MultiTrackBlockSorterOptions) error {
+		o.rule = rule
+		return nil
+	}
+}
+
+// WithMaxTimescaleDelay set the maximum allowed deley between tracks for a
+// given timescale.
+func WithMaxTimescaleDelay(maxTimescaleDelay int64) MultiTrackBlockSorterOption {
+	return func(o *MultiTrackBlockSorterOptions) error {
+		o.maxTimescaleDelay = maxTimescaleDelay
+		return nil
+	}
+}
+
 // NewMultiTrackBlockSorter creates BlockInterceptor, which sorts blocks on multiple tracks by timestamp.
 // The index of TrackEntry sorts blocks with the same timestamp.
 // Place the audio track before the video track to meet WebM Interceptor Guidelines.
```

Comment:
Would be better to add a comment for godoc to indicates that one of WithMaxDelayedPackets and WithMaxTimescaleDelay must be specified.
Also better to describe the behavior when both of them are specified.